torch/distributions/transforms.py

```diff
@@ -354,19 +355,34 @@ class AffineTransform(Transform):
 
     def __init__(self, loc, scale, event_dim=0, cache_size=0):
         super(AffineTransform, self).__init__(cache_size=cache_size)
-        self.loc, self.scale = broadcast_all(loc, scale)
+        self.loc = loc
+        self.scale = scale
```

Comment:
Why can’t we promote them to tensors here? Not having clearly typed attributes will be a constant source of bugs

Reply:
I don't think you need to know that exactly. If they are tensors, then keep them like it, and if they are scalars then we are actually in the process of adding support for implicit casting, so you can probably safely use the default tensor type (`torch.tensor(loc)`). See #5864.

Reply:
Oh great, are you're planning to add casting from e.g. `torch.FloatTensor` to `torch.cuda.DoubleTensor` before 0.4 release? If so, then I'll revert this 3-line change.

Reply:
After discussing offline I've reverted this change. I'll move this to a separate PR for further discussion.